components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ThrottlingLimitMappingUtil.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.publisher.v1.common.mappings;
+
+import org.wso2.carbon.apimgt.api.model.ThrottlingLimit;
+import org.wso2.carbon.apimgt.rest.api.publisher.v1.dto.APIOperationsThrottlingLimitDTO;
+
+/**
+ * This class is responsible for mapping APIOperationsThrottlingLimitDTOs into REST API throttling related manipulations
+ * and vice versa.
+ */
+public class ThrottlingLimitMappingUtil {
+
+    /**
+     * Manipulates ThrottlingLimit object from APIOperationsThrottlingLimitDTO
+     *
+     * @param dto Holds throttling limits relevant to the operations
+     * @return throttlingLimit object
+     */
+    public static ThrottlingLimit fromDTOToThrottlingLimit(APIOperationsThrottlingLimitDTO dto) {
+        ThrottlingLimit throttlingLimit = new ThrottlingLimit();
+        throttlingLimit.setRequestCount(dto.getRequestCount());
+        throttlingLimit.setUnit(dto.getUnit().toString());
```

Comment:
There can be an NPE if someone doesn't send the Unit.. shall we make both request count and unit as required?

Reply:
fixed with: https://github.com/wso2/carbon-apimgt/pull/11760/commits/e5916cf71524fd2dda951e046c65c7454b8faef4